activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -42,6 +42,7 @@ def initialize(scope)
       #    User.where.not(name: "Jon", role: "admin")
       #    # SELECT * FROM users WHERE name != 'Jon' AND role != 'admin'
       def not(opts, *rest)
+        opts = @scope.send(:sanitize_opts, opts)
```

Comment:
hmm we can't. `where!` is not called if it is called the following way: `where.not(params)`.